src/main/resources/templates/fragments/navbar.html

```diff
@@ -0,0 +1,85 @@
+<div th:fragment="navbar" id="navbar">
+    <nav class="navbar">
+        <div class="navbar-brand">
+            <a class="navbar-item" href="/">OurShop</a>
+        </div>
+
+        <!-- Navbar Links -->
+        <div class="navbar-links">
+            <a class="navbar-item" href="/">Home</a>
+            <a class="navbar-item" href="/categories">Categories</a>
+            <a class="navbar-item" href="/products">Products</a>
+        </div>
+
+
+        <button class="navbar-toggle" onclick="toggleNavbar()">☰</button>
+    </nav>
+
+    <style>
+        .navbar {
+            display: flex;
+            justify-content: space-between;
+            align-items: center;
+            background-color: #333;
+            padding: 0.5rem 1rem;
+            color: #fff;
+        }
+
+        .navbar a {
+            color: #fff;
+            text-decoration: none;
+            margin: 0 0.5rem;
+            padding: 0.5rem;
+        }
+
+        .navbar a:hover {
+            background-color: #555;
+            border-radius: 4px;
+        }
+
+        .navbar-brand {
+            font-weight: bold;
+        }
+
+        .navbar-links {
+            display: flex;
+        }
+
+        .navbar-toggle {
+            display: none;
+            font-size: 1.5rem;
+            background: none;
+            border: none;
+            color: #fff;
+            cursor: pointer;
+        }
+
+        @media (max-width: 750px) {
+            .navbar-links {
+                display: none;
+                flex-direction: column;
+                width: 100%;
+                background-color: #333;
+                position: absolute;
+                top: 3rem;
+                left: 0;
+                padding: 1rem 0;
+            }
+
+            .navbar-links.active {
+                display: flex;
+            }
+
+            .navbar-toggle {
+                display: block;
+            }
+        }
+    </style>
+
+    <script>
```

Comment:
the script can be also separated in a js file

Reply:
done

Reply:
done